pkg/pool-aclamm/contracts/AclAmmPool.sol

```diff
@@ -35,29 +45,24 @@ contract AclAmmPool is BalancerPoolToken, PoolInfo, Version, IBasePool, BaseHook
     // Invariant shrink limit: non-proportional remove cannot cause the invariant to decrease by less than this ratio.
     uint256 internal constant _MIN_INVARIANT_RATIO = 70e16; // 70%
 
+    SqrtQ0State private _sqrtQ0State;
     uint256 private _lastTimestamp;
-    uint256[] private _virtualBalances;
-
     uint256 private _c;
-    uint256 private _sqrtQ0;
     uint256 private _centernessMargin;
-
-    /// @dev Struct with data for deploying a new AclAmmPool.
-    struct AclAmmPoolParams {
-        string name;
-        string symbol;
-        string version;
-        uint256 increaseDayRate;
-        uint256 sqrtQ0;
-        uint256 centernessMargin;
-    }
+    uint256[] private _virtualBalances;
 
     constructor(
         AclAmmPoolParams memory params,
         IVault vault
-    ) BalancerPoolToken(vault, params.name, params.symbol) PoolInfo(vault) Version(params.version) {
+    )
+        BalancerPoolToken(vault, params.name, params.symbol)
+        PoolInfo(vault)
+        BasePoolAuthentication(vault, msg.sender)
+        Version(params.version)
+    {
         _setIncreaseDayRate(params.increaseDayRate);
-        _setSqrtQ0(params.sqrtQ0);
+
+        _sqrtQ0State.endSqrtQ0 = params.sqrtQ0;
```

Comment:
We should use the internal _setSqrtQ0 here, and pass 0 as start and endTime (or current timestamp), as well as params.sqrtQ0 as start and end value.

Doing that, we make sure the same event is emitted, which is better for offchain calculations.

Reply:
I created another event for constructor 

`event AclAmmPoolInitialized(uint256 increaseDayRate, uint256 sqrtQ0, uint256 centernessMargin);`